doc/spec/json.md

```diff
@@ -0,0 +1,76 @@
+# Docker Distribution JSON Canonicalization
+
+## Introduction
+
+To provide consistent content hashing of JSON objects throughout Docker
+Distribution APIs, a canonical JSON format has been defined. Adopting such a
+canonicalization also aids in caching JSON responses.
+
+## Rules
+
+Compliant JSON should conform to the following rules:
+
+1. All generated JSON should comply with [RFC
+   7159](http://www.ietf.org/rfc/rfc7159.txt).
+2. Resulting "JSON text" shall always be encoded in UTF-8.
```

Comment:
Is the objective of this line to guarantee that string contents can be meaningfully parsed as UTF-8?